src/lib.rs

```diff
@@ -1,3 +1,9 @@
+#![cfg_attr(target_arch = "spirv", no_std)]
```

Comment:
Shouldn't these be conditioned on the `std` feature, instead of `arch = spirv`? There's nothing spir-v specific here, and getting `no_std` working on non-spirv platforms seems useful to other folks.

Reply:
It could indeed be, though I did pick doing it on the arch here for now as this arch doesn't support std at all right now. But you are right if one wants to use no_std for other platforms (which is not supported now though), then one would use the `std` feature instead here. Could be cleaner also

Reply:
It would be nice to add general `no_std` support although the mechanism might be different for spirv and the general case. For the general case I was thinking of trying https://github.com/rust-lang/libm. That might not be a good solution for your use case though? There is a long standing issue to support `no_std` in `glam`, I've just never go around to it - see https://github.com/bitshifter/glam-rs/issues/5.

Reply:
would indeed be nice to have general no_std support, and depending on approach, if one does use core intrinsics like here + libm there is a potential of using the same implementation. 

because we talked a bit about using `libm` also and in the backend replace some of the functions with SPIRV intrinsics when available as a potential solution for implementing the SPIRV intrinsics that Rust core or LLVM does not expose.

but we'll see how this evolves, definitely surprisingly similar problems here on CPU and GPU no_std :)